src/Fixer/ClassNotation/FinalClassFixer.php

```diff
@@ -0,0 +1,113 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Fixer\ClassNotation;
+
+use PhpCsFixer\AbstractFixer;
+use PhpCsFixer\DocBlock\DocBlock;
+use PhpCsFixer\FixerDefinition\CodeSample;
+use PhpCsFixer\FixerDefinition\FixerDefinition;
+use PhpCsFixer\Tokenizer\Token;
+use PhpCsFixer\Tokenizer\Tokens;
+
+/**
+ * @author Filippo Tessarotto <zoeslam@gmail.com>
+ */
+final class FinalClassFixer extends AbstractFixer
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function getDefinition()
+    {
+        return new FixerDefinition(
+            'All classes must be final, except abstract ones and Doctrine entities.',
+            [
+                new CodeSample(
+                    '<?php
+class MyApp {}
+'
+                ),
+            ],
+            null,
+            'Risky when subclassing non-abstract classes.'
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isCandidate(Tokens $tokens)
+    {
+        return $tokens->isTokenKindFound(T_CLASS);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isRisky()
+    {
+        return true;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function applyFix(\SplFileInfo $file, Tokens $tokens)
+    {
+        $classes = array_keys($tokens->findGivenKind(T_CLASS));
+
+        while ($classIndex = array_pop($classes)) {
+            // ignore class if it is abstract, already final or anonymous
+            $prevToken = $tokens[$tokens->getPrevMeaningfulToken($classIndex)];
+            if ($prevToken->isGivenKind([T_ABSTRACT, T_FINAL, T_NEW])) {
+                continue;
+            }
+
+            // ignore class if it's a Doctrine Entity
+            $docToken = $tokens[$tokens->getPrevNonWhitespace($classIndex)];
+            if ($this->isDoctrineEntity($docToken)) {
+                continue;
+            }
+
+            $tokens->insertAt(
+                $classIndex,
+                [
+                    new Token([T_FINAL, 'final']),
+                    new Token([T_WHITESPACE, ' ']),
+                ]
+            );
+        }
+    }
+
+    /**
+     * @param Token $docToken
+     *
+     * @return bool
+     */
+    private function isDoctrineEntity(Token $docToken)
+    {
+        if (!$docToken->isGivenKind(T_DOC_COMMENT)) {
+            return false;
+        }
+
+        $doc = new DocBlock($docToken->getContent());
+        foreach ($doc->getAnnotations() as $annotation) {
+            $tag = strtolower($annotation->getTag()->getName());
+            if ('entity' === $tag || false !== stripos($annotation->getContent(), '@orm\entity')) {
```

Comment:
What about `@Entity`? I'm not overly familiar with Doctrine but when I used it for a project I was following [this](https://www.doctrine-project.org/projects/doctrine-orm/en/2.6/tutorials/getting-started.html) which uses `@Entity`